crates/control/src/kick_target_provider.rs

```diff
@@ -135,8 +136,11 @@ fn collect_kick_targets(
     let field_to_ground = ground_to_field.inverse();
 
     let is_own_kick_off = matches!(
-        filtered_game_controller_state.map(|x| x.game_state),
-        Some(FilteredGameState::Playing { kick_off: true, .. })
+        filtered_game_controller_state.map(|x| (x.game_state, x.game_phase)),
+        Some((
+            FilteredGameState::Playing { kick_off: true, .. },
+            GamePhase::Normal
```

Comment:
Probably matter of taste but I'd prefer just matching the struct rather than mapping it to a tuple first.

```rs
    let is_own_kick_off = matches!(
        filtered_game_controller_state,
        Some(FilteredGameControllerState {
            game_state: FilteredGameState::Playing { kick_off: true, .. },
            game_phase: GamePhase::Normal,
            ..
        })
    );
    let is_not_free_for_opponent = matches!(
        filtered_game_controller_state,
        Some(FilteredGameControllerState {
            opponent_game_state: FilteredGameState::Playing {
                ball_is_free: false,
                ..
            },
            ..
        })
    );
```

Reply:
Fine for me, changed to your suggestion